src/_app/use-cases/profile/my-profile.ts

```diff
@@ -0,0 +1,19 @@
+import { useAccessToken } from '~~/src/composables/auth/auth-cookies'
+
+export const useMyProfile = () => {
+  const pokeApi = usePokeApi()
+  const accessToken = useAccessToken()
+
+  const { data, refresh } = useAsyncData('profile/my', async () => {
+    return pokeApi.profile.getMyProfile({
+      meta: {
+        accessToken: accessToken.value,
```

Comment:
In the future, will there be an interceptor here?

Reply:
`before` will be a middleware with it's own `context` parameter